pkg/server/api/bundle/v1/service.go

```diff
@@ -161,17 +155,85 @@ func (s *Service) GetFederatedBundle(ctx context.Context, req *bundle.GetFederat
 		return nil, status.Errorf(codes.NotFound, "bundle for %q not found", req.TrustDomain)
 	}
 
-	b, err := applyMask(dsResp.Bundle, req.OutputMask)
+	b, err := api.BundleToProto(dsResp.Bundle)
 	if err != nil {
-		log.Errorf("Failed to apply mask: %v", err)
-		return nil, status.Errorf(codes.Internal, "failed to apply mask: %v", err)
+		log.WithError(err).Error("Failed to convert bundle")
+		return nil, status.Errorf(codes.Internal, "failed to convert bundle: %v", err)
 	}
 
+	applyBundleMask(b, req.OutputMask)
+
 	return b, nil
 }
 
 func (s *Service) BatchCreateFederatedBundle(ctx context.Context, req *bundle.BatchCreateFederatedBundleRequest) (*bundle.BatchCreateFederatedBundleResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "method BatchCreateFederatedBundle not implemented")
+	log := rpccontext.Logger(ctx)
+
+	results := []*bundle.BatchCreateFederatedBundleResponse_Result{}
+	for _, b := range req.Bundle {
+		td, err := spiffeid.TrustDomainFromString(b.TrustDomain)
+		if err != nil {
+			log.Errorf("Trust domain argument is not a valid SPIFFE ID: %q", b.TrustDomain)
```

Comment:
The trust domain string could be either the host part (e.g. example.org) or the SPIFFE ID URI (e.g. spiffe://example.org). We actually refer to "the name of the trust domain the bundle belongs to": https://github.com/spiffe/spire/blob/master/proto/spire-next/types/bundle.proto#L7
So I would just say in the error that the trust domain argument is not valid.